### PR TITLE
Update array push merging two arrays

### LINKED ISSUE #5000: Issue with "Array.prototype.push()": (short summary here please)
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push

#### What information was incorrect, unhelpful, or incomplete?
It seems incomplete or perhaps idiosyncratic to offer `Array.prototype.push.apply()` as a way to merge arrays without first giving an example of `push()` with destructuring assignment: `oneArray.push(...anotherArray)` 

#### Specific section or headline?
Merging two arrays using apply()

#### What did you expect to see?
`vegetables.push(...moreVegs)`

#### Did you test this? If so, how?
JavaScript console

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/array/push`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/array/push/index.html
* Last commit: https://github.com/mdn/content/commit/13360cbad46b4ce6457f80492d73d2efd252e8a9
* Document last modified: 2021-05-05T10:31:52.000Z

</details>

### PR DESCRIPTION
**Pull request by jadjoubran**
> What was wrong/why is this fix needed? (quick summary only)
Update merging two arrays example to use spread syntax and mention .concat instead of .apply

> Issue number (if there is an associated issue)

Closes #5000 

> Anything else that could help us review it

### CODE REVIEW

**Comment by Ryuno-Ki** [comment]
Do you happen to know, whether the former version did a deep or shallow copy?
Spread is only shallow-copying an Array.

So this could have slightly different semantics.

**Comment by wbamberg** [comment]
> Do you happen to know, whether the former version did a deep or shallow copy?
> Spread is only shallow-copying an Array.
> 
> So this could have slightly different semantics.

If I understand you right, this should show that the old version did a shallow copy:

```js
let beetroot = { color: "red", tasty: true }
let cabbage = { color: "green", tasty: true }

let a1 = [beetroot]
let a2 = [cabbage]

Array.prototype.push.apply(a1, a2)

a2[0].tasty = false

console.log(a1[1])  // Object { color: "green", tasty: false }
console.log(a2[0])  // Object { color: "green", tasty: false }
```

File: files/en-us/web/javascript/reference/global_objects/array/push/index.html [hunk 1]
```diff
@@ -71,24 +71,20 @@ <h3 id="Adding_elements_to_an_array">Adding elements to an array</h3>
...
 <h3 id="Merging_two_arrays">Merging two arrays</h3>
 
-<p>This example uses {{jsxref("Function.apply", "apply()")}} to push all elements from a
-  second array.</p>
-
-<p>Do <em>not</em> use this method if the second array (<code>moreVegs</code> in the
-  example) is very large because the maximum number of parameters that one function can
-  take is limited in practice. See {{jsxref("Function.apply", "apply()")}} for more
-  details.</p>
+<p>This example uses {{jsxref("Operators/Spread_syntax", "spread syntax")}} to push all elements from a
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/array/push/index.html:74]:
```suggestion
<p>This example uses {{jsxref("Operators/Spread_syntax", "spread syntax", "", "1")}} to push all elements from a
```

**Review by wbamberg — COMMENTED** [review]
I added a couple of commits:

* the jsxref call that wanted to link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax used "spread syntax" as the link text, which isn't an identifier and therefore shouldn't be in `code`. You can use a magic fourth argument to suppress this: https://github.com/mdn/yari/blob/main/kumascript/macros/jsxref.ejs#L10.
* merged from main to fix CI (thanks @peterbe for the quick fix!)

Thanks @jadjoubran for this contribution!

**Review by wbamberg — APPROVED** [review]

**Pull request merged**